components/calendar/src/ethiopic.rs

```diff
@@ -0,0 +1,204 @@
+// This file is part of ICU4X. For terms of use, please see the file
+// called LICENSE at the top level of the ICU4X source tree
+// (online at: https://github.com/unicode-org/icu4x/blob/main/LICENSE ).
+
+//! This module contains types and implementations for the Ethiopic calendar
+
+use crate::coptic::Coptic;
+use crate::iso::{Iso, IsoYear};
+use crate::julian::Julian;
+use crate::{
+    types, ArithmeticDate, Calendar, CalendarArithmetic, Date, DateDuration, DateDurationUnit,
+    DateTime, DateTimeError,
+};
+use core::marker::PhantomData;
+use tinystr::tinystr;
+
+#[derive(Copy, Clone, Debug, Hash, Default, Eq, PartialEq)]
+// The Ethiopic Calendar
+pub struct Ethiopic;
+
+#[derive(Copy, Clone, Debug, Hash, Eq, PartialEq)]
+pub struct EthiopicDateInner(ArithmeticDate<Ethiopic>);
+
+impl CalendarArithmetic for Ethiopic {
+    fn month_days(year: i32, month: u8) -> u8 {
+        if (1..=12).contains(&month) {
+            30
+        } else if month == 13 {
+            if Self::is_leap_year(year) {
+                6
+            } else {
+                5
+            }
+        } else {
+            0
+        }
+    }
+
+    fn months_for_every_year() -> u8 {
+        13
+    }
+
+    fn is_leap_year(year: i32) -> bool {
```

Comment:
Pretty sure this is wrong. Looking for the right method. Sources would help

Reply:
This is right

Code pointer: https://github.com/unicode-org/icu/blob/main/icu4j/main/classes/core/src/com/ibm/icu/util/CECalendar.java#L216

Reply:
Are we sure that their extended year is set according to the same zero year as we're using here?

Reply:
I think the best thing to do is just verify with some tests that compare to the Chrome implementation